pkg/kritis/review/review_test.go

```diff
@@ -0,0 +1,298 @@
+			isWebhook:         true,
+			attestations:      validAtts,
+			handledViolations: 0,
+			isAttested:        true,
+			shdAttestImage:    false,
+			shdErr:            false,
+		},
+		{
+			name:              "vulnz w/o attestation for Webhook shd handle voilations",
+			image:             vulnImage,
+			isWebhook:         true,
+			attestations:      []metadata.PGPAttestation{},
+			handledViolations: 1,
+			isAttested:        false,
+			shdAttestImage:    false,
+			shdErr:            true,
+		},
+		{
+			name:              "no vulnz w/o attestation for webhook shd add attestation",
+			image:             noVulnImage,
+			isWebhook:         true,
+			attestations:      []metadata.PGPAttestation{},
+			handledViolations: 0,
+			isAttested:        false,
+			shdAttestImage:    true,
+			shdErr:            false,
+		},
+		{
+			name:              "vulnz w attestation for cron shd handle vuln",
+			image:             vulnImage,
+			isWebhook:         false,
+			attestations:      validAtts,
+			handledViolations: 1,
+			isAttested:        true,
+			shdAttestImage:    false,
+			shdErr:            true,
+		},
+		{
+			name:              "vulnz w/o attestation for cron shd handle vuln",
+			image:             vulnImage,
+			isWebhook:         false,
+			attestations:      []metadata.PGPAttestation{},
+			handledViolations: 1,
+			isAttested:        false,
+			shdAttestImage:    false,
+			shdErr:            true,
+		},
+		{
+			name:              "no vulnz w/o attestation for cron shd verify attestations",
+			image:             noVulnImage,
+			isWebhook:         false,
+			attestations:      []metadata.PGPAttestation{},
+			handledViolations: 0,
+			isAttested:        false,
+			shdAttestImage:    false,
+			shdErr:            false,
+		},
+		{
+			name:              "no vulnz w attestation for cron shd verify attestations",
+			image:             noVulnImage,
+			isWebhook:         false,
+			attestations:      []metadata.PGPAttestation{{Signature: sigNoVuln, KeyId: "sec"}},
+			handledViolations: 0,
+			isAttested:        true,
+			shdAttestImage:    false,
+			shdErr:            false,
+		},
+	}
+	for _, tc := range tests {
+		th := violation.MemoryStrategy{
+			Violations:   map[string]bool{},
+			Attestations: map[string]bool{},
+		}
+		t.Run(tc.name, func(t *testing.T) {
+			cMock := &testutil.MockMetadataClient{
+				PGPAttestations: tc.attestations,
+			}
+			r := New(cMock, &th, tc.isWebhook, &Config{
+				validate: testValidate,
+				secret:   sMock,
+			})
+			if err := r.Review([]string{tc.image}, isps, nil); (err != nil) != tc.shdErr {
+				t.Fatalf("expected review to return error %t, actual error %s", tc.shdErr, err)
+			}
+			if len(th.Violations) != tc.handledViolations {
+				t.Fatalf("expected to handle %d violations. Got %d", tc.handledViolations, len(th.Violations))
+			}
+
+			if th.Attestations[tc.image] != tc.isAttested {
+				t.Fatalf("expected to get image attested: %t. Got %t", tc.isAttested, th.Attestations[tc.image])
```

Comment:
Prefer t.Errorf over t.Fatalf when possible (here and elsewhere)

Even after your test cases encounter a failure, they should keep going for as long possible in order to print out all of the failed checks in a single run. This way, someone who's fixing the failing test doesn't have to play whac-a-mole, fixing one bug and then re-running the test to find the next bug.

t.Fatal is usually only appropriate when some piece of test setup fails, without which you cannot run the test at all. In a table-driven test, t.Fatal is appropriate for failures that set up the whole test function before the test loop. Failures that affect a single entry in the test table, which make it impossible to continue with that entry, should be reported as follows: